### PR TITLE
Run `python setup.py sdist` first then `pip install`

### PR DESCRIPTION
**Pull request by harupy**
### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

I don't know if this works, but the spark docs say `python setup.py sdist` needs to be executed first.

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/ml-package-versions.yml [hunk 1]
```diff
@@ -310,7 +310,9 @@ spark:
       # -Pconnect -Phive is requried by spark connect
       ./build/mvn -Pconnect -Phive -DskipTests --no-transfer-progress clean package
       cd python
-      pip install '.[connect]'
+      python setup.py sdist
+      sdist=$(find dist -type f -name "*.tar.gz")
+      pip install "pyspark[connect] @ $sdist"
```
Comment (harupy) [inline mlflow/ml-package-versions.yml:315]:
```suggestion
      pip install "pyspark[connect] @ file://$sdist"
```

File: mlflow/ml-package-versions.yml [hunk 2]
```diff
@@ -310,7 +310,9 @@ spark:
       # -Pconnect -Phive is requried by spark connect
       ./build/mvn -Pconnect -Phive -DskipTests --no-transfer-progress clean package
       cd python
-      pip install '.[connect]'
+      python setup.py sdist
+      sdist=$(find dist -type f -name "*.tar.gz")
+      pip install "pyspark[connect] @ file://$sdist"
```
Comment (harupy) [inline mlflow/ml-package-versions.yml:315]:
```suggestion
      sdist=$(find dist -type f -name "*.tar.gz")
      pip install "$sdist[connect]"
```

File: tests/spark/test_spark_connect_model_export.py [hunk 3]
```diff
@@ -25,10 +25,10 @@ def _get_spark_connect_session():
     builder = SparkSession.builder.remote("local[2]").config(
         "spark.connect.copyFromLocalToFs.allowDestLocal", "true"
     )
-    if not Version(pyspark.__version__).is_devrelease:
-        builder.config(
-            "spark.jars.packages", f"org.apache.spark:spark-connect_2.12:{pyspark.__version__}"
-        )
+    builder.config(
```
[the remark this answers is not in the document]
Comment (harupy) [reply tests/spark/test_spark_connect_model_export.py:28]: thanks! let me try this.

File: mlflow/ml-package-versions.yml [hunk 4]
```diff
@@ -310,7 +310,9 @@ spark:
       # -Pconnect -Phive is requried by spark connect
       ./build/mvn -Pconnect -Phive -DskipTests --no-transfer-progress clean package
```
Comment (WeichenXu123) [inline mlflow/ml-package-versions.yml:311]: according to discussion, we should use `sbt` instead of mvn ?

Comment (harupy) [reply mlflow/ml-package-versions.yml:311]: Not in this PR.

**Comment by harupy** [comment]
Fixed by #9820

**Pull request closed without merging**